pkg/discovery/stitching/node_uuid.go

```diff
@@ -203,6 +205,50 @@ func (gce *gceNodeUUIDGetter) GetUUID(node *api.Node) (string, error) {
 	return result, nil
 }
 
+/**
+Input Vsphere.k8s.Node info:
+ spec:
+   // The systemUUID identifies a node uniquely in most older environments and
+   // matches that discovered by vsphere.
+   // In newer versions (eg. tanzu) the id discovered from insfrastructure is
+   // populated in the providerID.
+   // We observe both fields and if both are present we send both, else send only
+   // the information deciphered from providerID.
+   systemUUID: 4200C244-1E27-8473-AB44-999195DE924C
+   providerID: vsphere://4200c244-1e27-8473-ab44-999195de924c
+
+   //Note: the ids listed in the example above are same but can be reported different too.
+ Output:  4200c244-1e27-8473-ab44-999195de924c,4200c244-1e27-8473-ab44-999195de924c
+*/
+
+type vsphereNodeUUIDGetter struct {
+}
+
+func (v *vsphereNodeUUIDGetter) Name() string {
+	return "Vsphere"
+}
+
+func (v *vsphereNodeUUIDGetter) GetUUID(node *api.Node) (string, error) {
+	providerId := node.Spec.ProviderID
+	if !strings.HasPrefix(providerId, vspherePrefix) {
+		glog.Errorf("Not a valid Vspehere node uuid: %++v", node)
+		return "", fmt.Errorf("Invalid")
+	}
+
+	//1. Get the suffix, this is one of our ids:
+	//   4200c244-1e27-8473-ab44-999195de924c
+	stitchingID := providerId[len(vspherePrefix):]
+
+	//2. Check if node.Status.NodeInfo.SystemUUID is filled by provider
+	//   If it is, append that as another id.
+	//   4200c244-1e27-8473-ab44-999195de924c,4200c244-1e27-8473-ab44-999195de924c
+	if node.Status.NodeInfo.SystemUUID != "" {
```

Comment:
Nit: Are there a lot of cases where the `SystemUUID` is the same as the `ProviderID`, just like the example here?  If yes, then it may be worth an extra check to make sure they are different before adding it?

Reply:
Thanks makes sense. Updated!